rnd/autogpt_server/autogpt_server/data/graph.py

```diff
@@ -112,47 +128,70 @@ async def get_node(node_id: str) -> Node | None:
 
 
 async def get_graph_ids() -> list[str]:
-    return [graph.id for graph in await AgentGraph.prisma().find_many()]  # type: ignore
+    return [
+        graph.id
+        for graph in await AgentGraph.prisma().find_many(where={"is_template": False})
+    ]  # type: ignore
+
+
+async def get_template_meta() -> list[GraphMeta]:
+    templates = await AgentGraph.prisma().find_many(
+        where={"is_template": True},
+    )
+    if not templates:
+        return []
 
+    return [GraphMeta.from_db(template) for template in templates]  # type: ignore
 
-async def get_graph(graph_id: str) -> Graph | None:
-    graph = await AgentGraph.prisma().find_unique(
-        where={"id": graph_id},
+
+async def get_graph(graph_id: str, is_template: bool = False) -> Graph:
```

Comment:
graph_id will always be either template or non_template, so the `is_template` is redundant ?

Reply:
Since `/graphs/{graph_id}` and `/templates/{graph_id}` are separate routes, I have slightly changed the behavior of `get_template` to not return `404 Graph #{graph_id} is not a template.` but `404 Template #{graph_id} not found.`. For this we don't need to actually retrieve the `graph`, so we can just do the filtering in `data.graph:get_graph` rather than implementing it twice in both API endpoint methods.